packages/filebrowser/src/listing.ts

```diff
@@ -1646,8 +1628,8 @@ export class DirListing extends Widget {
   private _clipboard: string[] = [];
   private _manager: IDocumentManager;
   private _softSelection = '';
-  private _selection: { [key: string]: boolean } = Object.create(null);
-  private _renderer: DirListing.IRenderer;
+  protected _selection: { [key: string]: boolean } = Object.create(null);
+  protected _renderer: DirListing.IRenderer;
```

Comment:
Also making these two protected so that they can be accessed in subclasses

Reply:
Would it make sense to completely expose the renderer (public) as for: https://github.com/jupyterlab/jupyterlab-git/pull/885 ?

Reply:
I guess so.

I start to believe this renderer should not be mandatory for DirListing subclasses. The current IRenderer is really bound to the way things are done in the core filebrowser currently (a list of files in one directory), which makes it very difficult to subclass and extend for e.g. a file tree logic.